src/Samples/NetCoreServer/Program.cs

```diff
@@ -15,25 +15,21 @@ static void Main(string[] args)
             host.Run();
         }
 
+        // Listen on 8088 for http, and 8443 for https, 8089 for NetTcp.
         public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
             .UseKestrel(options => {
-                options.ListenLocalhost(8088);
-                options.ListenLocalhost(8443, listenOptions =>
+                options.ListenLocalhost(Startup.HTTP_PORT);
```

Comment:
AllowSynchronousIO flag should be set here also I think.

Reply:
I'm digging in to that now to find the conditions when that's the case. We only synchronously read in 2 situations.  
1. Using StreamedMode transfer
2. Using BufferedMode and the client does NOT send a ContentLength header. We read a single byte of the request body to see if there is a request body. I've almost got a fix ready where AllowSynchronousIO only needs to be set when using StreamedMode.

Reply:
When you have the answer, I will fix these up to be consistent.